packages/protocol/src/market/modules/LiquidityModule.sol

```diff
@@ -643,6 +715,108 @@ contract LiquidityModule is ReentrancyGuardUpgradeable, ILiquidityModule {
                 deltaCollateral: deltaCollateral
             })
         );
+
+        // Notice: closing the trade position after the event is emitted to have both events show the valid intermediate state
+        if (position.kind == IFoilStructs.PositionKind.Trade && closeTrade) {
+            _closeTradePosition(epoch, position, tradeSlippage);
+        }
+    }
+
+    function _closeTradePosition(
+        Epoch.Data storage epoch,
+        Position.Data storage position,
+        uint256 tradeSlippage
+    ) internal {
+        uint256 initialPrice = epoch.getReferencePrice();
+        int256 deltaCollateralLimit = -int256(
+            position.depositedCollateralAmount.mulDecimal(1e18 - tradeSlippage)
+        );
+
+        Trade.QuoteOrTradeInputParams memory inputParams = Trade
+            .QuoteOrTradeInputParams({
+                oldPosition: position,
+                initialSize: position.positionSize(),
+                targetSize: 0,
+                deltaSize: -position.positionSize(),
+                isQuote: false
+            });
+
+        // Do the trade
+        Trade.QuoteOrTradeOutputParams memory outputParams = Trade.quoteOrTrade(
+            inputParams
+        );
+
+        position.vEthAmount = outputParams.position.vEthAmount;
+        position.vGasAmount = outputParams.position.vGasAmount;
+        position.borrowedVEth = outputParams.position.borrowedVEth;
+        position.borrowedVGas = outputParams.position.borrowedVGas;
+        position.depositedCollateralAmount = outputParams
+            .position
+            .depositedCollateralAmount;
+
+        // Ensures that the position only have single side tokens
+        position.rebalanceVirtualTokens();
+
+        // 1. Confirm no vgas tokens
+        if (position.vGasAmount > 0) {
+            // Notice. This error should not happen. If it's here it means something went wrong
+            revert Errors.InvalidData("Cannot close position with vGas tokens");
+        }
+        if (position.borrowedVGas > 0) {
+            // Notice. This error should not happen. If it's here it means something went wrong
+            revert Errors.InvalidData(
+                "Cannot close position with borrowed vGas tokens"
+            );
+        }
```

Comment:
these checks seem unnecessary since the code paths leading to this code ensure this can't happen but 🤷🏽